### ISSUE TITLE
Rule Request: Use where in loop instead of if

### ISSUE THREAD

**Issue by kimdv** [realm/SwiftLint#1228]
For better readable code in my opinion it would be nice with a where rule for loops like. 

I think it will make it more readable and there is less lines. Of course if there is multiple if or and if else it should be ignored

```swift
for user in users where user.id == "1" {
// Awesome code
}
```

instead of
```swift
for user in users {
    if user.id == "1" {
        // Awesome code
   }
}
```

**Comment by abagmut** [comment]
get false positive for:
```
var name = ""
for (index, value) in lines.enumerated()
{
	if index > 0
	{
		name += ", "
	}
	name += value
}
```

**Comment by marcelofabri** [comment]
There's already an open issue for that, as well as a PR.

**Comment by Coeur** [comment]
For reference:
* issues reporting the false positives: https://github.com/realm/SwiftLint/pull/1387, https://github.com/realm/SwiftLint/issues/1404
* pull-request fixing the false positives: https://github.com/realm/SwiftLint/pull/1393
